source/MRMesh/MRTriangleIntersection.h

```diff
@@ -72,6 +72,9 @@ bool doTrianglesIntersect(
     if ( abde * acde < 0 )
         return true; // DE segment penetrates triangle ABC since points B and C are at distinct sides of ADE
 
+    if ( abdf == 0 && acde == 0 )
+        return true; // ...
```

Comment:
some comment would be highly appreciated. Let us draw a picture first, and when write it.